benches/benches/vm.rs

```diff
@@ -41,25 +40,48 @@ where
                 db_txn
             };
 
-            let mut elapsed_time = Duration::default();
-            for _ in 0..iters {
+            let final_time;
+            loop {
+                // Measure the total time to revert the VM to the initial state.
+                // It should always do the same things regardless of the number of
+                // iterations because we use a `diff` from the `VmBenchPrepared` initialization.
                 let start = std::time::Instant::now();
-                match instruction {
-                    Instruction::CALL(call) => {
-                        let (ra, rb, rc, rd) = call.unpack();
-                        vm.prepare_call(ra, rb, rc, rd).unwrap();
-                    }
-                    _ => {
-                        black_box(vm.instruction(*instruction).unwrap());
+                for _ in 0..iters {
+                    vm.reset_vm_state(diff);
```

Comment:
Is this actually measuring anything? It seems like this would essentially be a noop if we haven't executed any instructions yet 🤔

Reply:
During setup of the bench we run instruction one time to take a `diff` 
<img width="1225" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/b226ff87-9f5b-404c-a8b8-7ce696e9fa02">

Reply:
Have we run the benchmarks to see how much of an improvement this is?

Reply:
Yes, now it produces more stable results 
<img width="818" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/50818700-7171-441f-b95a-92791c5cbfd0">
<img width="697" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/472aa595-1b0d-4e05-8130-a324538c8014">

Reply:
that's great!